src/zeep/xsd/visitor.py

```diff
@@ -802,10 +802,11 @@ def visit_union(self, node, parent):
         """
         # TODO
         members = node.get('memberTypes')
+        default_namespace = node.nsmap.get(None)
         types = []
         if members:
             for member in members.split():
-                qname = as_qname(member, node.nsmap, self.document._target_namespace)
+                qname = as_qname(member, node.nsmap, default_namespace or self.document._target_namespace)
```

Comment:
Btw, when loading the XSD XML from the union testcases in Altova XMLSpy I get validations errors, some types cannot be resolved. I don't know who's in the wrong here, Altova or the XSD in the testcase ...